sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/FeedResponseDiagnostics.java

```diff
@@ -131,4 +145,17 @@ public void addClientSideRequestStatistics(List<ClientSideRequestStatistics> req
         clientSideRequestStatisticsList.addAll(requestStatistics);
     }
 
+    public Duration getFeedResponseCreationLatency() {
+        return feedResponseCreationLatency.get();
+    }
+
+    public void recordFeedResponseCreationLatency(Duration feedResponseCreationLatency) {
+
+        if (this.feedResponseCreationLatency.get() != Duration.ZERO) {
+            throw new IllegalStateException("The feed response latency has been recorded already.");
```

Comment:
I think we should just ignore the fact that it has already been recorded. Throwing error while capturing diagnostics is not a right design in my opinion. I think diagnostics capture should be a silent operation (in the background).